Bottom: 2b43779631801f4b3e938f9cbd668a034ef37155
Top:    edc2eccee3101ff6344b3712e4e566153c75d8e6
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-10 16:34:32 +0000

Refresh of revert-fix-centerwindow-drc-0

---

diff --git a/src/hid.h b/src/hid.h
index 69e4541..650a24b 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -529,7 +529,6 @@ typedef enum
 
 #define HID_SC_CENTER_IN_VIEWPORT                  3
 
-#define HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER 4
 
       hidval (*add_timer) (void (*func) (hidval user_data_),
 			   unsigned long milliseconds_, hidval user_data_);
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index a4ece3a..d49421b 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -39,7 +39,6 @@ pan_common (GHidPort *port)
   ghidgui->adjustment_changed_holdoff = FALSE;
 
   ghid_port_ranges_changed();
-  gdk_window_process_all_updates ();
 }
 
 static void
@@ -304,9 +303,7 @@ ghid_set_crosshair (int x, int y, int action)
     }
 
   if (action != HID_SC_PAN_VIEWPORT &&
-      action != HID_SC_WARP_POINTER &&
-      action != HID_SC_CENTER_IN_VIEWPORT &&
-      action != HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER)
+      action != HID_SC_WARP_POINTER)
     return;
 
   /* Find out where the drawing area is on the screen. gdk_display_get_pointer
@@ -315,36 +312,9 @@ ghid_set_crosshair (int x, int y, int action)
    */
   gdk_window_get_origin (gtk_widget_get_window (gport->drawing_area),
                          &offset_x, &offset_y);
-
   display = gdk_display_get_default ();
-  screen = gdk_display_get_default_screen (display);
 
   switch (action) {
-
-    case HID_SC_CENTER_IN_VIEWPORT:
-
-      // Center the viewport on the crosshair
-      ghid_pan_view_abs (gport->crosshair_x - gport->view.width / 2,
-                         gport->crosshair_y - gport->view.height / 2,
-                         0, 0);
-
-      break;
-
-    case HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER:
-
-      // Center the viewport on the crosshair
-      ghid_pan_view_abs (gport->crosshair_x - gport->view.width / 2,
-                         gport->crosshair_y - gport->view.height / 2,
-                         0, 0);
-
-      // Warp pointer to crosshair
-      ghid_pcb_to_event_coords (gport->crosshair_x, gport->crosshair_y,
-                                &widget_x, &widget_y);
-      gdk_display_warp_pointer (display, screen, widget_x + offset_x,
-                                widget_y + offset_y);
-
-      break;
-
     case HID_SC_PAN_VIEWPORT:
       /* Pan the board in the viewport so that the crosshair (who's location
        * relative on the board was set above) lands where the pointer is.
diff --git a/src/hid/gtk/gui-drc-window.c b/src/hid/gtk/gui-drc-window.c
index 14421b3..9cb578f 100644
--- a/src/hid/gtk/gui-drc-window.c
+++ b/src/hid/gtk/gui-drc-window.c
@@ -169,8 +169,7 @@ row_activated_cb (GtkTreeView *view, GtkTreePath *path,
   if (violation == NULL)
     return;
 
-  CenterDisplay (violation->x_coord, violation->y_coord, true);
-  gtk_window_present (GTK_WINDOW (gport->top_window));
+  CenterDisplay (violation->x_coord, violation->y_coord);
 }
 
 enum
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 2cb3393..a04ab62 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -3535,16 +3535,6 @@ lesstif_set_crosshair (int x, int y, int action)
 
     }
 
-  if (action == HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER)
-    {
-      fprintf (
-          stderr,
-          "warning:%s:%i: HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER not "
-          "implemented in this HID, using HID_SC_WARP_POINTER instead\n",
-          __FILE__,
-          __LINE__ );
-      action = HID_SC_WARP_POINTER;
-    }
   if (action == HID_SC_PAN_VIEWPORT)
     {
       Window root, child;
diff --git a/src/misc.c b/src/misc.c
index f98aeff..5bfbe98 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -835,18 +835,10 @@ void
 CenterDisplay (Coord X, Coord Y, bool warp_pointer)
 {
   Coord save_grid = PCB->Grid;
-
   PCB->Grid = 1;
-
   if (MoveCrosshairAbsolute (X, Y))
     notify_crosshair_change (true);
-
-  if (warp_pointer)
-    gui->set_crosshair (Crosshair.X, Crosshair.Y,
-                        HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER );
-  else
-    gui->set_crosshair (Crosshair.X, Crosshair.Y, HID_SC_CENTER_IN_VIEWPORT);
-
+  gui->set_crosshair (Crosshair.X, Crosshair.Y, HID_SC_WARP_POINTER);
   PCB->Grid = save_grid;
 }
